sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -629,7 +629,7 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase with ExecutionE
   }
 
   def failedToCompileMsg(e: Exception): String = {
```

Comment:
do we still need this function?

Reply:
Yes, it is used in `QueryExecutionErrors.internalCompilerError(e)`  and `QueryExecutionErrors.compilerError(e)`

Reply:
shall we also change them to put the original exception as the cause?

Reply:
@cloud-fan No, those looks ok. This PR is to complete https://issues.apache.org/jira/browse/SPARK-47580